main.cpp

```diff
@@ -42,12 +42,12 @@ struct T
 
 struct Comparison
 {
-    T* compare(T* a, T* b)
+    T* compare(T& a, T& b)
     {
-        if(a != nullptr && b != nullptr)
+        //if(a != nullptr && b != nullptr)
         {
-            if( a->value < b->value ) return a;
-            if( a->value > b->value ) return b;
+            if( a.value < b.value ) return &a;
+            if( a.value > b.value ) return &b;
         }
```

Comment:
Don't just comment it out.  Delete the curly braces as well and fix the indentation
- [ ] revise